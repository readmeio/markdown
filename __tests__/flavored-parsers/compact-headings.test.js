import { mdast } from '../../index';

describe('Compact headings', () => {
  it('can parse compact headings', () => {
    const heading = `#Compact Heading`;
    expect(mdast(heading, { settings: { position: true } })).toMatchSnapshot();
  });

  it('can parse headings that are not compact', () => {
    const heading = `# Non-compact Heading`;
    expect(mdast(heading, { settings: { position: true } })).toMatchSnapshot();
  });
});
