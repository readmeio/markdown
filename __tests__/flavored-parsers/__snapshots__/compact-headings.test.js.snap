// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compact headings can parse compact headings 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 33,
              "line": 1,
              "offset": 32,
            },
            "indent": Array [],
            "start": Object {
              "column": 18,
              "line": 1,
              "offset": 17,
            },
          },
          "type": "text",
          "value": "Compact Heading",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 1,
          "line": 2,
          "offset": 17,
        },
        "indent": Array [
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 2,
      "line": 3,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`Compact headings can parse headings that are not compact 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 22,
              "line": 1,
              "offset": 21,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 1,
              "offset": 2,
            },
          },
          "type": "text",
          "value": "Non-compact Heading",
        },
      ],
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 2,
      "line": 3,
      "offset": 24,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
